services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/docker_api.py

```diff
@@ -544,28 +560,24 @@ async def update_scheduler_data_label(scheduler_data: SchedulerData) -> None:
             with attempt:
                 try:
                     # fetch information from service name
-                    service_inspect = await client.services.inspect(
-                        scheduler_data.service_name
-                    )
+                    service_inspect = await client.services.inspect(service_name)
+                    service_inspect = await client.services.inspect(service_name)
                     service_version = service_inspect["Version"]["Index"]
                     service_id = service_inspect["ID"]
                     spec = service_inspect["Spec"]
 
-                    spec["Labels"][
-                        DYNAMIC_SIDECAR_SCHEDULER_DATA_LABEL
-                    ] = scheduler_data.as_label_data()
+                    updated_spec = update_spec_data(deepcopy(spec))
 
                     await client._query_json(  # pylint: disable=protected-access
                         f"services/{service_id}/update",
                         method="POST",
-                        data=json.dumps(clean_map(spec)),
+                        data=json.dumps(clean_map(updated_spec)),
                         params={"version": service_version},
                     )
                 except aiodocker.exceptions.DockerError as e:
                     if not (
                         e.status == status.HTTP_404_NOT_FOUND
-                        and e.message
-                        == f"service {scheduler_data.service_name} not found"
+                        and e.message == f"service {service_name} not found"
```

Comment:
do we really have to rely on the message contents? this is highly unstable don't you think?, same for the check below

Reply:
From what I can tell, the  message comes directly from the docker daemon. Both cases are some error state. 
The only source of instability is docker changes their API (which can happen and I guess you were referring to this).

If you have an alternative I'd gladly take it into consideration. For now I can add some tests for the API to guarantee that if these change we can detect it.

Reply:
but why not only checking the error code?

Reply:
I'm not sure to what you are referring. If that is the http status code, that is not enough.

Reply:
I am refering to ```e.message == f"service {service_name} not found"```
and in which case does the docker API return a 404 for something else than service not found error?